doc/4-instrument/fm-esfm.md

```diff
@@ -0,0 +1,119 @@
+
+### routing controls
+
+- **Output Level (OL)**: sets the output level from this operator to the audio output (0 to 7).
+  - 7 is the loudest level and 1 is the softest, while 0 disables audio output.
+  - this output scaling factor is applied after TL and envelope scaling have been performed.
+  - a change of one unit is about 6 dB.
+- **Modulation Input Level (MI)**: sets the modulation level from the previous operator to this operator (0 to 7).
+  - 7 is the strongest level and 1 is the weakest, while 0 disables modulation.
+  - for operator 1, since there's no previous operator, this controls the **feedback level**.
+  - this modulation scaling factor is applied after the previous operator's TL and envelope scaling have been performed, but is unaffected by OL above.
+  - a change of one unit is about 6 dB.
+
+### table of OL and MI values for obtaining common algorithms
+
+note: MI1 is not included as it is the feedback level.
+
+| algorithm                      | OL1 | MI2 | OL2 | MI3 | OL3 | MI4 | OL4 |
+|--------------------------------|:---:|:---:|:---:|:---:|:---:|:---:|:---:|
+| OPN algorithm **0**            |  0  |  7  |  0  |  7  |  0  |  7  |  7  |
+| OPN algorithm **4**            |  0  |  7  |  7  |  0  |  0  |  7  |  7  |
+| OPN algorithm **6**            |  0  |  7  |  7  |  0  |  7  |  0  |  7  |
+| OPN algorithm **7**            |  7  |  0  |  7  |  0  |  7  |  0  |  7  |
+| OPL3 algorithm **1**           |  7  |  0  |  0  |  7  |  0  |  7  |  7  |
+| OPL3 algorithm **3**           |  7  |  0  |  0  |  7  |  7  |  0  |  7  |
+| OPL3 algorithm **1** (variant) |  0  |  7  |  0  |  7  |  7  |  0  |  7  |
+
+### fixed frequency mode
+
+each operator has a Fixed Frequency mode. once enabled, the operator runs at the specified frequency regardless of the note.
+
+## macros
+
+these macros allow you to control several parameters of FM per tick.
+
+## OP1-OP4 Macros
+
+most parameters are listed above.
+
+### envelope delay macro tricks
+
+due to a quirk in how the envelope delay feature works, the **Envelope Delay** macro can control the operator's key-on status for a limited amount of time after a note is played. a value of 0 represents key-on, while a value of 7 represents key-off.
+
+note that the macro cannot exceed 659.1 ms in length - anything beyond that will be treated as a value of 0.
+
+### operator arpeggio and pitch
+
+among the available macros are **Op. arpeggio** and **Op. pitch**. these work like the **Arpeggio** and **Pitch** macros featured below, but are applied individually over the operator, overriding the **Arpeggio**/**Pitch** macros respectively.
+
+## Macros
+
+- **Volume**: volume sequence.
+- **OP4 Noise Mode**: controls special noise effects for operator 4 (as described before).
+- **Arpeggio**: pitch sequence.
+- **Panning**: toggles output on left and right channels.
+  - note that each operator also has its own pan controls, which get masked by this global pan control.
```

Comment:
Sorry, this seems a little unclear. Masked how?

Reply:
For instance, if an operator is set to only output on the right channel, and you set the global pan to only output on the left channel, it'll mask all of the operators' pan controls with the global pan setting - hence that operator that was set to output on the right channel won't output on any of the channels.

Reply:
Now that you explain it, it seems a lot more obvious. I think this language will work fine!